datafusion/physical-plan/src/projection.rs

```diff
@@ -60,15 +56,12 @@ pub struct ProjectionExec {
     input: Arc<dyn ExecutionPlan>,
     /// The output ordering
     output_ordering: Option<Vec<PhysicalSortExpr>>,
-    /// The columns map used to normalize out expressions like Partitioning and PhysicalSortExpr
-    /// The key is the column from the input schema and the values are the columns from the output schema
-    columns_map: HashMap<Column, Vec<Column>>,
+    /// The mapping used to normalize expressions like Partitioning and
+    /// PhysicalSortExpr. The key is the expression from the input schema
+    /// and the value is the expression from the output schema.
+    projection_mapping: Vec<(Arc<dyn PhysicalExpr>, Arc<dyn PhysicalExpr>)>,
```

Comment:
This is another pattern that has appeared more than once (also in GroupBy) -- maybe it could be `ProjectionMapping` (which could also be made into a struct w/ a constructor)

Reply:
BTW I made a PR to do this here -- https://github.com/apache/arrow-datafusion/pull/8033 and it worked out well. Once this PR is merged I'll rebase it and get it ready for review

Reply:
Here is a similar one for EquivalenceClass -- https://github.com/apache/arrow-datafusion/pull/8034 that might be more of a conflict magnet so leaving it for a while